### DISCUSSION TITLE
Multi UE support

### DISCUSSION THREAD

**Discussion by arrowtem** [srsran/srsRAN_Project#160] [Ideas & Suggestions]
Hello, do you have in plans implementing multi ue support like amarisoft ue simulator ?

**Comment by frankist** [comment]
Not in the near future. For now, we will use COTS UEs and the amarisoft UE simulator.

**Reply by arrowtem** [reply]
How is it possible to run 1 on sdr multiple UE’s , If their positions of PUCCH resources could overlap in freq domain ( the same with RACH)
Were you testing 64 UE’s connected and performing traffic test ? IPerf? Was enb able to handle it ?
Thanks much for answer !

**Reply by frankist** [reply]
We have tested 64 UEs with ping and iperf using amarisoft UE simulator. The amarisoft UE simulator (and other simulators as well) allows emulating several UEs attaching to a gNB using a single radio front-end. I am not sure I understood your initial question regarding PRACH and PUCCH. During the Random Access (RA) procedure, each UE picks a PRACH preamble at random that it will use to generate the PRACH sequence to send to the gNB. The gNB is able to detect more than one PRACH at once, due to the orthogonality proprieties of these sequences. There is a non-zero chance that two or more UEs pick the same PRACH preamble, which will likely lead to some of them failing their RA procedure and having to repeat it again. This is all expected according to the NR TS, and it is handled by our gNB. For the PUCCH, our gNB scheduler ensures that there are no collisions. This is done via assigning different PUCCH resources (e.g. slot, symbol, PRB, cyclic shift) to different UEs. All that the UE simulator has to do is decode the PUCCHs for each of the simulated UEs separately.

With COTS, we were only able to test with the few phones we had available in the lab.

**Reply by arrowtem** [reply]
I was trying to run 4 UE’s in 4G , but srsENB was giving resources for them with collision ( same prb for 4 UE’s at same tti), do I need to configure srsENB another way ? Right now UE’s can’t send SR or harq at same time

**Comment by andrepuschmann** [comment]
Hey @arrowtem - please note that srsRAN Project only deals with the RAN and doesn't implement any UE functionality at all.
